### PR TITLE
docs: update sphinx theme

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----
Replaces #4263

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -21,47 +21,145 @@
...
+ogp_site_url = "https://canonical-snapcraft.readthedocs-hosted.com"
+ogp_site_name = project
+ogp_image = "https://assets.ubuntu.com/v1/253da317-image-document-ubuntudocs.svg"
+
+# Update with the favicon for your product
+html_favicon = "sphinx-resources/.sphinx/_static/favicon.png"
+
+html_context = {
+    "discourse": "https://forum.snapcraft.io",
+    "github_url": "https://github.com/canonical/snapcraft",
```
Comment (lengau) [inline docs/conf.py:34]:
```suggestion
    "github_url": "https://github.com/snapcore/snapcraft",
```

**Review by lengau — APPROVED** [review]
Looks good with the one suggested URL fix

**Comment by mr-cal** [comment]
`tox run -e build-docs` doesn't work for me on this branch, looks like CI is failing for a similar reason on `tox run -e lint-docs`

**Comment by sergiusens** [comment]
I mostly picked this up verbatim, @lengau would you know what's going on?

**Comment by lengau** [comment]
I think @mr-cal's PR #4375 got it :-)

**Comment by mr-cal** [comment]
Yep, that and 3 minor changes here.  I think we're good to go.
- https://github.com/snapcore/snapcraft/pull/4367/commits/f3e238c8ccffa6e5b36c538e05aa0fabc19ca4ec
- https://github.com/snapcore/snapcraft/pull/4367/commits/9670947beaa7f8f1de127c5d2f2e73bf041ae7a3
- https://github.com/snapcore/snapcraft/pull/4367/commits/d34090209e07e1c51a2ef27dcce7ae98d7f35e91

**Review by mr-cal — APPROVED** [review]

**Pull request merged**